src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -99,17 +115,33 @@ pub struct WorkUnitStore {
 
 #[derive(Default)]
 pub struct WorkUnitInnerStore {
```

Comment:
It feels like this struct is holding a fair amount of redundant information now. Perhaps the `WorkunitRecord`s could become the Graph's `NodeWeight` type to avoid the extra HashMap? That would look like:
```
type WorkunitGraph = DiGraph<WorkunitRecord, (), u32>;
```

Reply:
Agreed, although I'd like to refactor `WorkUnitInnerStore` in a follow-up commit.